src/p2p/net_peerlist_boost_serialization.h

```diff
@@ -134,10 +134,11 @@ namespace boost
       a & port;
       a & length;
 
-      if (length > net::tor_address::buffer_size())
+      const size_t buffer_size = net::tor_address::buffer_size();
+      if (length > buffer_size)
```

Comment:
For reference, we reported this bug upstream https://gcc.gnu.org/bugzilla/show_bug.cgi?id=90825